### PR TITLE
RUNO: Migrate codebase from ExpressJS to NestJS

### PR DESCRIPTION
**Pull request by Runo-saduwa**
#### What does this PR achieve?
This PR Migrates the codebase to NestJS. The file changes associated with this PR is as result of the of the following actions:
- Installs NestJS, its dependencies and boilerplates
- Deletes conflicting files
- Updates test command in .circleci
- Updates test script to also run give test coverage reports
- Adds new folders to the src directories - validators, constants, controllers, routes, repositories, helpers, models
- Updates .gitignore
- ensures all tests pass and coverage reports comes out successfully
- uninstall dormant dependencies 
#### Any background context?
The project was initially set up with express and sequelize, but the team took into consideration the future prospects of the project and we decided to replace the pioneering technologies with more modern and powerful technologies that will help our app to be more resilient and to support easy scalability.
#### Manual Testing Steps
- Fetch branch
- Install dependencies with `npm install`
- run `npm start` to start the server in `http://localhost:3000`
- run tests with `npm run test`
#### Relevant Trello Tickets
- [Ticket 1](some.com)

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,14 @@
+DB_USER=postgres
```
Comment (chidioguejiofor) [inline .env]: Add .env to gitignore

Comment (chidioguejiofor) [reply .env]: This file should not be seen here

**Review by chidioguejiofor — COMMENTED** [review]
This looks good except the .env. Also, I think you should have a:
- src/controllers folder to hold all controllers
- src/services folder to hold all services

File: .env (hunk 1 above)
Comment (Runo-saduwa) [reply .env]: Oops, my mistake.

**Review by chidioguejiofor — APPROVED** [review]

**Pull request merged**